app/models/account.rb

```diff
@@ -576,6 +581,10 @@ def independent_mapping_rules_enabled?
 
   private
 
+  def reset_counter_cache
+    update_column(:bought_cinstances_count, bought_cinstances.size) if persisted?
```

Comment:
why is this necessary? 😕 shouldn't it do this automatically without overriding this method?

Reply:
@macejmic

Reply:
`counter_cache` is not defined for `bought_cinstances` so yes, it has to be done manually, that's the concept of the manual counter cache.